src/coreclr/tools/dotnet-pgo/PgoCompareMethodFlowGraph.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ILCompiler;
+using Internal.Pgo;
+
+namespace Microsoft.Diagnostics.Tools.Pgo
+{
+    internal class PgoCompareMethodFlowGraph
+    {
+        public bool ProfilesHadBasicBlocks { get; init; }
+        public bool ProfilesHadEdges { get; init; }
+        public PgoCompareMethodBasicBlock EntryBasicBlock { get; init; }
+        public List<PgoCompareMethodBasicBlock> BasicBlocks { get; init; }
+
+        public long TotalBlockCount1 => BasicBlocks.Sum(bb => bb.BlockCount1);
+        public long TotalBlockCount2 => BasicBlocks.Sum(bb => bb.BlockCount2);
+        public long TotalEdgeCount1 => BasicBlocks.Sum(bb => bb.Edges.Sum(e => e.Value.Count1));
+        public long TotalEdgeCount2 => BasicBlocks.Sum(bb => bb.Edges.Sum(e => e.Value.Count2));
+
+        public double ComputeBlockOverlap()
```

Comment:
You might also look into the Spearman rank correlation as a measure of similarity.

Reply:
I'll look into it.